cmake/kokkos_compiler_id.cmake

```diff
@@ -31,16 +31,33 @@ IF(INTERNAL_HAVE_COMPILER_NVCC)
   SET(KOKKOS_CXX_COMPILER_VERSION ${TEMP_CXX_COMPILER_VERSION} CACHE STRING INTERNAL FORCE)
 ENDIF()
 
-IF(KOKKOS_CXX_COMPILER_ID STREQUAL Cray)
 
-  # SET nvcc's compiler version.
+# Check if the compiler is nvcc (which really means nvcc_wrapper).
+IF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+  # The Cray compiler reports as Clang to most versions of CMake
+  EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
+                  COMMAND grep Cray
+                  COMMAND wc -l
+                  OUTPUT_VARIABLE INTERNAL_HAVE_CRAY_COMPILER
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+  IF (INTERNAL_HAVE_CRAY_COMPILER) #not actually Clang
+    SET(KOKKOS_CLANG_IS_CRAY TRUE)
+  ENDIF()
+ENDIF()
+
+IF(KOKKOS_CXX_COMPILER_ID STREQUAL Cray OR KOKKOS_CLANG_IS_ACTUALLY_CRAY)
```

Comment:
Where is `KOKKOS_CLANG_IS_ACTUALLY_CRAY` defined? Should that be `KOKKOS_CLANG_IS_CRAY`?

Reply:
Type-o. Fixed.